MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,227 @@
+                                .DescendantNodes().OfType<ClassDeclarationSyntax>()
+                        where (string) cd.Identifier.Value == "MoreEnumerable"
+                        from md in cd.DescendantNodes().OfType<MethodDeclarationSyntax>()
+                        let mn = (string) md.Identifier.Value
+                        where md.ParameterList.Parameters.Count > 0
+                        && md.ParameterList.Parameters.First().Modifiers.Any(m => (string)m.Value == "this")
+                        && md.Modifiers.Any(m => (string)m.Value == "public")
+                        && md.AttributeLists.SelectMany(al => al.Attributes).All(a => a.Name.ToString() != "Obsolete")
+                        && includePredicate(mn)
+                        && !excludePredicate(mn)
+                        select md
+                };
+
+            var indent = new string('\x20', 4);
+            var indent2 = indent + indent;
+            var indent3 = indent2 + indent;
+
+            var imports =
+                from nss in new[]
+                {
+                    new []
+                    {
+                        "System",
+                        "System.CodeDom.Compiler",
+                        "System.Collections.Generic",
+                    },
+                    usings.AsEnumerable(),
+                }
+                from ns in nss
+                select indent + $"using {ns};";
+
+            var classes =
+                from f in q
+                from md in f.Methods
+                group md by (string) md.Identifier.Value into g
+                select new
+                {
+                    Name = g.Key,
+                    Overloads =
+                        from md in g
+                        select
+                            MethodDeclaration(md.ReturnType, md.Identifier)
+                                .WithAttributeLists(md.AttributeLists)
+                                .WithModifiers(md.Modifiers)
+                                .WithTypeParameterList(md.TypeParameterList)
+                                .WithConstraintClauses(md.ConstraintClauses)
+                                .WithParameterList(md.ParameterList)
+                                .WithExpressionBody(
+                                    ArrowExpressionClause(
+                                        InvocationExpression(
+                                            MemberAccessExpression(
+                                                SyntaxKind.SimpleMemberAccessExpression,
+                                                IdentifierName("MoreEnumerable"),
+                                                IdentifierName(md.Identifier)),
+                                            ArgumentList(SeparatedList(
+                                                from p in md.ParameterList.Parameters
+                                                select Argument(IdentifierName(p.Identifier)))))
```

Comment:
>  will roslyn automatically add a `@` marker if necessary?

So no and I guess [`CodeDomProvider.CreateValidIdentifier`](https://docs.microsoft.com/en-us/dotnet/api/system.codedom.compiler.codedomprovider.createvalididentifier?view=netframework-4.5.2) is still our friend then.